File: solutions/elixir/darts/2/lib/darts.ex
defmodule Darts do
  alias Enum

  @type position :: {number, number}

  @spec score(position) :: integer
  def score(position) do
    d = distance(position)

    cond do
      d <= 1 -> 10
      d <= 5 -> 5
      d <= 10 -> 1
      true -> 0
    end
  end

  defp distance({x, y}) do
    :math.sqrt(:math.pow(0 - x, 2) + :math.pow(0 - y, 2))
  end
end

File: solutions/fsharp/darts/2/Darts.fs
// Active Pattern version
module Darts

let distance x y = sqrt (x * x + y * y)

let (|Inner|Middle|Outer|Outside|) (d: float) =
    match d with
    | d when d <= 1 -> Inner
    | d when d <= 5 -> Middle
    | d when d <= 10 -> Outer
    | _ -> Outside

let score x y =
    match distance x y with
    | Inner -> 10
    | Middle -> 5
    | Outer -> 1
    | Outside -> 0
